client/client.go

```diff
@@ -103,6 +104,8 @@ func (c client) try(ctx context.Context, request Request, cancelFunc context.Can
 
 	for _, fm := range request.failManagers {
 		if err := fm.Check(resp); err != nil {
+			ioutil.ReadAll(resp.Body)
```

Comment:
A more explicit way of doing so would be to use Copy  and Discard : `io.Copy(ioutil.Discard, resp.Body)`